cli/azd/cmd/provision.go

```diff
@@ -206,6 +214,29 @@ func (p *provisionAction) Run(ctx context.Context) (*actions.ActionResult, error
 			}
 		}
 
+		//if user don't have access to openai
+		errorMsg := err.Error()
```

Comment:
I think we can help the user more here.

1. Once they create the service they won't be able to just come back to azd and run provision.  They will need to reconfigure azd to use that service by providing the name, location, and resource group as parameters to azd.
2. I think we should investigate a better message and a better follow through for how they can be successful when facing this error.
3. I believe there is a CLI that allows them to accept terms, at least they were working on it. So maybe point users to that?

Reply:
@hemarina 
- Can you check with `Erin Dormier` if any CLI is available for the same. Doubt if they have any.
- Can't we check for the specific error code `SpecialFeatureOrQuotaIdRequired` and `ResourceKindRequireAcceptTerms` instead of its messages which might be dynamic based on inputs, resource types.

Reply:
> Once they create the service they won't be able to just come back to azd and run provision. They will need to reconfigure azd to use that service by providing the name, location, and resource group as parameters to azd.

I don't think we should try to have the user create the resource they want to use in the portal and then have us try to adopt it in this manner.  Instead, I think we should have the user create a dummy resource in the portal (so they can do the consent flow once) and then have them delete that resource and then `azd provision` will work, since the consent has happened.  Does it not behave in this manner today, @jongio?  My impression was once they had consented once their entire subscription was allowed to create these resources, even via templates.

I would love if we can make that be the experience we start with, and then once there's a programmatic way to offer consent, update what @hemarina has here (in a separate PR) to prompt the user to consent inside `azd`.  Once that works we can also consider moving the consent to the start of the flow, by looking at the resource types we might allocate.  But I think that can come later?  Giving a good actionable message about how to use the portal one time to do the consent flow seems like the best we can do now?